interlok-core/src/main/java/com/adaptris/core/ConfigurableEventHandler.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2015 Adaptris Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package com.adaptris.core;
+
+import com.adaptris.annotation.AdapterComponent;
+import com.adaptris.annotation.AutoPopulated;
+import com.adaptris.annotation.ComponentProfile;
+import com.adaptris.annotation.InputFieldDefault;
+import com.adaptris.annotation.InputFieldHint;
+import com.adaptris.core.util.Args;
+import com.adaptris.core.util.LifecycleHelper;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamImplicit;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * <p>
+ * Configurable implementation of <code>EventHandler</code> that allows the user to
+ * send events to different producers based on rules. Rules are composed of an event matcher
+ * which checks whether an event should be sent by it's standalone producer. If there are
+ * no matching rules, the default configured producer is used.
+ * </p>
+ *
+ * @config default-event-handler
+ *
+ */
+@XStreamAlias("configurable-event-handler")
+@AdapterComponent
+@ComponentProfile(summary = "Sends MessageLifecycleEvents to a destination based on the matching rule, or default if no match", tag = "base,events")
```

Comment:
Not just MessageLifecycleEvents, but any Events (which is MLE + all of the various ALEs), isn't that right?

Reply:
Yes that is correct, I will update it to just Events